### PR TITLE
Catalyst installers

### PR DESCRIPTION
**Pull request by mcmigliore**

### CODE REVIEW

**Comment by mcmigliore** [comment]
@rupertnash Here are the scripts I used.  
I know this is probably very dirty but I have zero knowledge with the module system.  
Basically, I sourced the script `modules.sh` and run `install_pv.sh` (must be run first!), `install_ttk.sh` and `install_wfa.sh`.

File: paraview/install_pv.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  git submodule update --init
+  cd ..
+fi
+
+cd paraview-build
+cmake -DSUPERBUILD_PROJECT_PARALLELISM=8 \
+      -DPARAVIEW_BUILD_EDITION=CATALYST_RENDERING \
+      -Dparaview_SOURCE_SELECTION=5.8.1-RC1 \
+      -DENABLE_python=ON \
+      -DENABLE_python3=ON \
```
Comment (mcmigliore) [inline paraview/install_pv.sh:18]: We use the ParaView `superbuild` here. It builds ParaView with its dependencies (most notably python3, osmesa, ospray, nlohmann_json)

File: paraview/install_pv.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      -DENABLE_paraviewweb=ON \
+      -DENABLE_osmesa=ON \
+      -Dmesa_USE_SWR=OFF \
```
Comment (mcmigliore) [inline paraview/install_pv.sh:21]: GLIBC on Archer is too old, this is necessary

File: ttk/install_ttk.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Build TTK
+if [ ! -d ttk-source ]; then
+  git clone https://github.com/topology-tool-kit/ttk.git ttk-source
+  mkdir ttk-build
+  cd ttk-source
+  git checkout a26adce85a53a7581dfd4fc7bef96305db8a9383
+  rm -rf paraview/WRLExporter/ core/vtk/ttkWRLExporter/
```
Comment (mcmigliore) [inline ttk/install_ttk.sh:9]: This is currently required. A fix in TTK will be merged soon: https://github.com/topology-tool-kit/ttk/pull/451

File: install_wfa.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+if [ ! -d wfa-source ]; then
+  git clone https://git.firesponse.es/external/vestec.git wfa-source
+  cd wfa-source
+  git checkout 8fce0a845836968e2658a8c64c58f39fba77c652
+  cd ..
+  mkdir wfa-build
+fi
+
+cd wfa-build
+cmake -DUSE_CATALYST=ON \
```
Comment (mcmigliore) [inline install_wfa.sh:13]: This is the CMake variable that will trigger Catalyst adaptor build in the simulation code

File: modules.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+module swap PrgEnv-cray PrgEnv-gnu
+#module add anaconda/2.2.0-python3
+module add cmake/3.16.0
+module add cray-netcdf/4.6.1.3
+module add cray-hdf5/1.10.2.0
+module add boost
+module swap gcc gcc/7.3.0
+module swap cray-mpich cray-mpich/7.7.4
+
+export CRAYPE_LINK_TYPE=dynamic
```
Comment (mcmigliore) [inline modules.sh:10]: I'm not sure if it is required...

File: paraview/install_pv.sh (hunk 1 above)
Comment (rupertnash) [reply paraview/install_pv.sh:18]: Can we use the version of python I installed as a module? It's version 3.8.3

**Comment by rupertnash** [comment]
I am going to pull your scripts into the same layout as the others for consistency

File: paraview/install_pv.sh (hunk 1 above)
Comment (Meakk) [reply paraview/install_pv.sh:18]: Yes, you should be able to use your installed python version. Add `-DUSE_SYSTEM_python3=ON`

Comment (mcmigliore) [reply paraview/install_pv.sh:18]: Yes, you should be able to use your installed python version. Add `-DUSE_SYSTEM_python3=ON`

Comment (rupertnash) [reply paraview/install_pv.sh:18]: Fail :(

```
[ 73%] Generating CXX resource object /home/d170/d170/nashvest/app-install/paraview/build/paraview-build/superbuild/openimagedenoise/build/weights/rt_hdr_alb_nrm.cpp
  File "/home/d170/d170/nashvest/app-install/paraview/build/paraview-build/superbuild/openimagedenoise/src/scripts/resource_to_cpp.py", line 33
    with open(in_path, 'rb') as in_file, open(out_path, 'w') as out_file:
                                       ^
SyntaxError: invalid syntax
```
A quick dig shows that the super build isn't passing the correct value of `PYTHON_INTERP` to the configure step of openimagedenoise. Any ideas?

Comment (mcmigliore) [reply paraview/install_pv.sh:18]: I remember now that this was the exact reason why I have not used your python version... Sorry, no idea!

File: ttk/install_ttk.sh (hunk 3 above)
Comment (mcmigliore) [reply ttk/install_ttk.sh:9]: this is merged now so we can bump the commit hash and remove these lines.

Comment (mcmigliore) [reply ttk/install_ttk.sh:9]: Actually, according to @julien-tierny, it seems that the current dev branch is not stable so stick with this commit for now.

File: install_wfa.sh [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# Build WFA
+if [ ! -d wfa-source ]; then
+  git clone https://git.firesponse.es/external/vestec.git wfa-source
+  cd wfa-source
+  git checkout 8fce0a845836968e2658a8c64c58f39fba77c652
```
Comment (mcmigliore) [inline install_wfa.sh:7]: Need to be changed to 7bb97e12ea2fbc1f7758b91c1a05230025187561

File: install_wfa.sh [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  cd ..
+  mkdir wfa-build
+fi
+
+cd wfa-build
+cmake -DUSE_CATALYST=ON \
+      -DCMAKE_INSTALL_PREFIX=/home/d170/d170/shared/sw/fireapp/catalyst \
+      -DCMAKE_INSTALL_RPATH="\$ORIGIN/../lib:/home/d170/shared/sw/paraview/5.8.1-RC1/lib:/home/d170/shared/sw/gdal/3.1.0/lib" \
```
Comment (mcmigliore) [inline install_wfa.sh:15]: `\$ORIGIN/../lib:/home/d170/shared/sw/paraview/5.8.1-RC1/lib` is not needed anymore with the static adaptor lib.

**Pull request closed without merging**